### PR TITLE
Add java service instance

### PR DESCRIPTION
**Pull request by mbfrahry**
This PR adds java service instance to the sdk and adds another test to the database service instance.

```
make testacc TEST=./java TESTARGS="-run=TestAccServiceInstanceLifeCycle"
==> Checking that code complies with gofmt requirements...
ORACLE_ACC=1 go test -v ./java -run=TestAccServiceInstanceLifeCycle -timeout 120m
=== RUN   TestAccServiceInstanceLifeCycle
--- PASS: TestAccServiceInstanceLifeCycle (4023.93s)
PASS
ok  	github.com/hashicorp/go-oracle-terraform/java	4023.944s

make testacc TEST=./database TESTARGS="-run=TestAccServiceInstanceCloudStorage"
==> Checking that code complies with gofmt requirements...
ORACLE_ACC=1 go test -v ./database -run=TestAccServiceInstanceCloudStorage -timeout 120m
=== RUN   TestAccServiceInstanceCloudStorage
--- PASS: TestAccServiceInstanceCloudStorage (2289.91s)
PASS
ok  	github.com/hashicorp/go-oracle-terraform/database	2289.922s
```

### CODE REVIEW

**Comment by mbfrahry** [comment]
Also we are reading in the cloud storage username and password from the environment when not specified

File: database/database_client.go [hunk 1]
```diff
@@ -10,7 +10,6 @@ import (
 )
 
 const DB_ACCOUNT = "/Database-%s"
-const DB_USERNAME = "/Database-%s:%s"
```
Comment (grubernaut) [inline database/database_client.go]: Should this change be included in here?

File: database/database_client.go [hunk 2]
```diff
@@ -58,10 +57,6 @@ func (c *DatabaseClient) executeRequest(method, path string, body interface{}) (
 	return resp, nil
 }
 
-func (c *DatabaseClient) getUserName() string {
```
Comment (grubernaut) [inline database/database_client.go]: Same here

File: java/java_client.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	if body != nil {
+		req.Header.Set("Content-Type", "application/vnd.com.oracle.oracloud.provisioning.Service+json")
+	}
+	// Log the request before the authentication header, so as not to leak credentials
+	c.client.DebugLogString(debugReqString)
+
+	// Set the authentiation headers
+	req.Header.Add(AUTH_HEADER, *c.authHeader)
+	req.Header.Add(TENANT_HEADER, *c.client.IdentityDomain)
+	c.client.DebugLogString(fmt.Sprintf("Req (%+v)", req))
```
Comment (grubernaut) [inline java/java_client.go]: This will leak their auth header. Recommend printing the debug string, _then_ adding in the auth header

Comment (mbfrahry) [inline java/java_client.go]: Oooo good catch.

**Pull request merged**